src/plugins/discord/fetcher.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+
+import * as Model from "./models";
+import sortBy from "lodash.sortby";
+
+type fetchEndpoint = (endpoint: string) => Promise<any>;
+type FetchOptions = {|
+  guildId: number,
```

Comment:
On second though. I think it would be better if the `guildId` is not part of the Fetcher's options.
Instead, let's keep this at the `discordFetchPlan` level. Because the Fetcher should simply wrap API calls and accept it as an argument when needed. It's part of the mirroring process to know which guildId we've set as "our server".

Reply:
My rationale for this `options` object: 

```js
type FetchOptions = {|
  guildId: number,
  membersLimit: number,
  messagesLimit: number,
  reactionsLimit: number,
|};

```

These will be constants for the duration of a `Fetchers` use-case. I don't think we should parameterize variables that won't change, and it makes it easier to reason about the function parameters that _will_ change by keeping constants out of function signatures. E.g., we're not going to use the same `Fetcher` to fetch different `guildId`s.  It also reduces the surface area where mistakes could occur by providing these constants in one place (at the classes' initialization) instead of several (whenever we need to call a method that requires `guildId`). 

Not a strong opinion however. If we wanted to remove the `guildId` from the `options` object, I'd think we want to do the same with the `limit` variables as well. 

What do you think?